### DISCUSSION TITLE
Can Docker run inside a Firecracker VM?

### DISCUSSION THREAD

**Discussion by WeeJeWel** [firecracker-microvm/firecracker#3313] [Q&A]
I've tried installing Docker inside an Ubuntu microVM, but got stuck on network errors.

Is it actually feasible to run Docker containers inside a Firecracker VM?

**Comment by chwzr** [comment]
with the latest image from the getting started guid it will work just out of the box. 

I also had network problems with older rootfs images. 
Its very likely that running this inside the guest ```apt reinstall linux-modules-$(uname -r) -y``` will help.

**Comment by bchalios** [comment]
Docker should work just fine. It sounds as if your problem is networking setup. Have you followed the steps here: https://github.com/firecracker-microvm/firecracker/blob/main/docs/network-setup.md for properly setting up networking with internet access?

**Comment by ShadowCurse** [comment]
There is a https://github.com/firecracker-microvm/firecracker/discussions/4038 with more info
